packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,125 +211,560 @@ class CupertinoAlertDialog extends StatelessWidget {
   ///    section when it is long.
   final ScrollController actionScrollController;
 
-  @override
-  Widget build(BuildContext context) {
+  Widget _buildContent() {
     final List<Widget> children = <Widget>[];
 
     if (title != null || content != null) {
-      final Widget titleSection = new _CupertinoAlertTitleSection(
+      final Widget titleSection = new _CupertinoAlertContentSection(
         title: title,
         content: content,
         scrollController: scrollController,
       );
       children.add(new Flexible(flex: 3, child: titleSection));
-      // Add padding between the sections.
-      children.add(const Padding(padding: EdgeInsets.only(top: 8.0)));
     }
 
+    return new Container(
+      color: _kDialogColor,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: children,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container(
+      height: 0.0,
+    );
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return actionSection;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final double textScaleFactor = MediaQuery.of(context).textScaleFactor;
```

Comment:
I'm not sure what you mean about rebuilding, but the reason I didn't duplicate that condition in a bunch of places is because I don't think we want a bunch of non-accessibility widgets having an internal understanding of a convention that means we're in accessibility mode.  

The notion that text scale factor > X means accessibility mode is very much an iOS accessibility implementation detail and I don't think dialog widgets should be knowledgeable about it.

Eventually this question shouldn't even be answered within dialog.dart - we should have public API support to ask this question.  But accessibility isn't there yet, so this is a half-measure that keeps responsibilities separate.